server/chat/backend/agent/tools/gitlab_tool.py

```
@@ -0,0 +1,576 @@
+                user_dir = base_dir / user_id
+                if user_dir.exists():
+                    session_dirs = [d for d in user_dir.iterdir() if d.is_dir() and d.name.startswith('session_')]
+                    if session_dirs:
+                        terraform_dir = str(max(session_dirs, key=lambda x: x.stat().st_mtime))
+
+        if not terraform_dir or not Path(terraform_dir).exists():
+            return json.dumps({"status": "error", "error": "No Terraform files found to commit"})
+
+        terraform_files = []
+        for tf_file in Path(terraform_dir).glob("*.tf"):
+            with open(tf_file, 'r') as f:
```

Ruff (0.15.12)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)